--- a/Makefile.am
+++ b/Makefile.am
@@ -25,7 +25,6 @@
 	imagereader \
 	imagereader/src \
 	it \
-	lsmashsource \
 	msmoosh \
 	mvtools \
 	nnedi3 \
--- a/configure.ac
+++ b/configure.ac
@@ -70,7 +70,6 @@
 PKG_CHECK_MODULES([libavformat], [libavformat])
 PKG_CHECK_MODULES([libavutil], [libavutil])
 PKG_CHECK_MODULES([libswscale], [libswscale])
-PKG_CHECK_MODULES([liblsmash], [liblsmash])
 PKG_CHECK_MODULES([fftw3f], [fftw3])
 PKG_CHECK_MODULES([SNDFILE], [sndfile])
 
@@ -103,7 +102,6 @@
                  imagereader/Makefile \
                  imagereader/src/Makefile \
                  it/Makefile \
-                 lsmashsource/Makefile \
                  msmoosh/Makefile \
                  rawsource/Makefile \
                  retinex/Makefile \
