--- a/src/Deblock.cpp
+++ b/src/Deblock.cpp
@@ -294,7 +294,11 @@ static void VS_CC deblockFree(void *instanceData, VSCore *core, const VSAPI *vsa
 }
 
 static void VS_CC deblockCreate(const VSMap *in, VSMap *out, void *userData, VSCore *core, const VSAPI *vsapi) {
+#if __GNUC__ >= 5
     DeblockData d{};
+#else
+    DeblockData d;
+#endif
     int err;
 
     d.node = vsapi->propGetNode(in, "clip", 0, nullptr);
@@ -395,7 +399,11 @@ static void VS_CC deblockCreate(const VSMap *in, VSMap *out, void *userData, VSC
         return;
     }
 
+#if __GNUC__ >= 5
     DeblockData * data = new DeblockData{ std::move(d) };
+#else
+    DeblockData * data = new DeblockData(std::move(d));
+#endif
 
     vsapi->createFilter(in, out, "Deblock", deblockInit, deblockGetFrame, deblockFree, fmParallel, 0, data, core);
 
