--- configure.orig	2016-10-12 20:44:51.000000000 +0200
+++ configure	2016-11-01 16:37:38.109930726 +0100
@@ -491,8 +491,7 @@
 for tool in $TOOLS; do
     cat >> config.mak2 << EOF
 cli/${tool}${EXT}: cli/${tool}.o $OBJ_TOOLS $STATICLIB $SHAREDLIB
-	\$(CC) \$(CFLAGS) \$(LDFLAGS) -o \$@ \$< $OBJ_TOOLS -llsmash \$(LIBS)
-	-@ \$(if \$(STRIP), \$(STRIP) \$@)
+	\$(CCLD_silent)\$(CC) \$(CFLAGS) \$(LDFLAGS) -o \$@ \$< $OBJ_TOOLS -llsmash \$(LIBS)
 
 EOF
 done
