--- Makefile.orig	2017-03-05 19:11:49.000000000 +0100
+++ Makefile	2017-06-12 17:01:12.853141517 +0200
@@ -1,3 +1,10 @@
+ifneq ($(V),1)
+CCLD_silent   = @ echo '  CCLD     '$@;
+CC_silent     = @ echo '  CC       '$@;
+AR_silent     = @ echo '  AR       '$@;
+RANLIB_silent = @ echo '  RANLIB   '$@;
+endif
+
 # Makefile for L-SMASH
 
 # note:
@@ -10,27 +17,21 @@
 
 OBJS = $(SRCS:%.c=%.o)
 
-SRC_ALL = $(SRCS) $(SRC_TOOLS)
+SRC_ALL = $(SRCS)
 
 #### main rules ####
 
-.PHONY: all lib install install-lib clean distclean dep depend
-
-all: $(STATICLIB) $(SHAREDLIB) $(TOOLS)
+all: $(STATICLIB) $(SHAREDLIB)
 
 lib: $(STATICLIB) $(SHAREDLIB)
 
 $(STATICLIBNAME): $(OBJS)
-	$(AR) rc $@ $^
-	$(RANLIB) $@
+	$(AR_silent)$(AR) rc $@ $^
+	$(RANLIB_silent)$(RANLIB) $@
 	-@ $(if $(STRIP), $(STRIP) -x $@)
 
 $(SHAREDLIBNAME): $(OBJS)
-	$(LD) $(SO_LDFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
-	-@ $(if $(STRIP), $(STRIP) -x $@)
-ifneq ($(SLIB_CMD),)
-	$(SLIB_CMD)
-endif
+	$(CCLD_silent)$(CCLD) $(SO_LDFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
 ifeq ($(SHAREDLIBNAME), liblsmash.so.$(MAJVER))
 	ln -sf $(SHAREDLIBNAME) liblsmash.so
 endif
@@ -40,11 +41,9 @@
 include config.mak2
 
 %.o: %.c .depend config.h
-	$(CC) -c $(CFLAGS) -o $@ $<
+	$(CC_silent)$(CC) -c $(CFLAGS) -o $@ $<
 
 install: all install-lib
-	install -d $(DESTDIR)$(bindir)
-	install -m 755 $(TOOLS) $(DESTDIR)$(bindir)
 
 install-lib: liblsmash.pc lib
 	install -d $(DESTDIR)$(includedir)
