--- a/ax_check_cl.m4
+++ b/ax_check_cl.m4
@@ -34,21 +34,17 @@
 # the Macro.  You need not follow the terms of the GNU General Public
 # License when using or distributing such scripts.
 #
+
+# djcj <djcj@gmx.de>: I made some changes for VapourSynth on Linux
+# Original source(s): http://wili.cc/blog/opencl-m4.html
+#                     http://wili.cc/blog/entries/opencl-m4/ax_check_cl.m4
+# Backup:             https://gist.github.com/darealshinji/71dadd3157a94fd905bd6382c4b761bb
+
 AC_DEFUN([AX_CHECK_CL],
 [AC_REQUIRE([AC_CANONICAL_HOST])dnl
 AC_REQUIRE([AC_PATH_X])dnl
 AC_REQUIRE([AC_PROG_SED])dnl
-AC_REQUIRE([ACX_PTHREAD])dnl
 
-AC_LANG_PUSH([C])
-AX_LANG_COMPILER_MS
-AS_IF([test X$ax_compiler_ms = Xno],
-      [CL_CFLAGS="${PTHREAD_CFLAGS}"; CL_LIBS="${PTHREAD_LIBS} -lm"])
-
-#
-# Use x_includes and x_libraries if they have been set (presumably by
-# AC_PATH_X).
-#
 AS_IF([test X$no_x != Xyes],
       [AS_IF([test -n "$x_includes"],
              [CL_CFLAGS="-I$x_includes $CL_CFLAGS"])]
@@ -60,8 +56,6 @@
 AC_CHECK_HEADERS([CL/cl.h OpenCL/cl.h])
 CPPFLAGS=$ax_save_CPPFLAGS
 
-AC_CHECK_HEADERS([windows.h])
-
 m4_define([AX_CHECK_CL_PROGRAM],
           [AC_LANG_PROGRAM([[
 # if defined(HAVE_WINDOWS_H) && defined(_WIN32)
@@ -79,14 +73,16 @@
 AC_CACHE_CHECK([for OpenCL library], [ax_cv_check_cl_libcl],
 [ax_cv_check_cl_libcl=no
 case $host_cpu in
-  x86_64) ax_check_cl_libdir=lib64 ;;
-  *)      ax_check_cl_libdir=lib ;;
+  x86_64) ax_check_cl_libdir=lib64 ;
+          ax_check_cl_libdir_multi=x86_64-linux-gnu ;;
+  *)      ax_check_cl_libdir=lib ;
+          ax_check_cl_libdir_multi=i386-linux-gnu ;;
 esac
 ax_save_CPPFLAGS=$CPPFLAGS
 CPPFLAGS="$CL_CFLAGS $CPPFLAGS"
 ax_save_LIBS=$LIBS
 LIBS=""
-ax_check_libs="-lOpenCL -lCL"
+ax_check_libs="-lOpenCL -lCL /usr/lib/$ax_check_cl_libdir_multi/libOpenCL.so.1"
 for ax_lib in $ax_check_libs; do
   AS_IF([test X$ax_compiler_ms = Xyes],
         [ax_try_lib=`echo $ax_lib | $SED -e 's/^-l//' -e 's/$/.lib/'`],
@@ -94,26 +90,16 @@
   LIBS="$ax_try_lib $CL_LIBS $ax_save_LIBS"
 AC_LINK_IFELSE([AX_CHECK_CL_PROGRAM],
                [ax_cv_check_cl_libcl=$ax_try_lib; break],
-               [ax_check_cl_nvidia_flags="-L/usr/$ax_check_cl_libdir/nvidia" LIBS="$ax_try_lib $ax_check_cl_nvidia_flags $CL_LIBS $ax_save_LIBS"
-               AC_LINK_IFELSE([AX_CHECK_CL_PROGRAM],
-                              [ax_cv_check_cl_libcl="$ax_try_lib $ax_check_cl_nvidia_flags"; break],
-                              [ax_check_cl_dylib_flag='-dylib_file /System/Library/Frameworks/OpenCL.framework/Versions/A/Libraries/libCL.dylib:/System/Library/Frameworks/OpenCL.framework/Versions/A/Libraries/libCL.dylib' LIBS="$ax_try_lib $ax_check_cl_dylib_flag $CL_LIBS $ax_save_LIBS"
-                              AC_LINK_IFELSE([AX_CHECK_CL_PROGRAM],
-                                             [ax_cv_check_cl_libcl="$ax_try_lib $ax_check_cl_dylib_flag"; break])])])
+               [ax_check_cl_nvidia_flags="-L/usr/$ax_check_cl_libdir/nvidia -L/usr/lib/$ax_check_cl_libdir_multi/nvidia"
+                LIBS="$ax_try_lib $ax_check_cl_nvidia_flags $CL_LIBS $ax_save_LIBS"])
 done
 
-AS_IF([test "X$ax_cv_check_cl_libcl" = Xno -a X$no_x = Xyes],
-      [LIBS='-framework OpenCL'
-      AC_LINK_IFELSE([AX_CHECK_CL_PROGRAM],
-                     [ax_cv_check_cl_libcl=$LIBS])])
-
 LIBS=$ax_save_LIBS
 CPPFLAGS=$ax_save_CPPFLAGS])
 
 AS_IF([test "X$ax_cv_check_cl_libcl" = Xno],
       [no_cl=yes; CL_CFLAGS=""; CL_LIBS=""],
       [CL_LIBS="$ax_cv_check_cl_libcl $CL_LIBS"])
-AC_LANG_POP([C])
 
 AC_SUBST([CL_CFLAGS])
 AC_SUBST([CL_LIBS])
